src/display/api.js

```diff
@@ -2916,6 +2916,7 @@ class WorkerTransport {
         annotationStorage: this.annotationStorage.serializable,
         filename: this._fullReader?.filename ?? null,
       })
+      .catch(ex => warn(`Error when saving the document: ${ex.message}`))
```

Comment:
Is this a left-over from debugging, or do you actually intend to print a warning in the API-code?

As-is I believe that the `saveDocument´ method will now *always* return a resolved Promise,  which isn't intended if/when failure occurs (for example, it'll break the fallback in the viewer).

In any case, can you please just log a warning [here instead](https://github.com/mozilla/pdf.js/blob/c914e9f0a631b92ed81d1a685b42ae088adefcdc/web/app.js#L1016) using `console.error`.